### DISCUSSION TITLE
Filter on first view displayed doesn't work until I go back to it from another view

### DISCUSSION THREAD

**Discussion by armandgrillet** [community/community#5770] [Projects and Issues] [answered]
Hi, I have a board with two views and the first one includes a filter `-label:"type/epic"`.

When visiting the board for the first time, the first view will show issues with a label `type/epic` despite the feature being enabled. If I click on the second view then the first view, the filter applies itself correctly.

When loading the page for the first time:
<img width="1100" alt="PS 2021-09-23 à 14 52 51" src="https://user-images.githubusercontent.com/2117580/134510071-b32df709-52e8-4797-abaa-d28a9eba9145.png">

Then going on the second view:
<img width="1087" alt="PS 2021-09-23 à 14 53 18" src="https://user-images.githubusercontent.com/2117580/134510136-68e5fe34-18c8-4178-9085-5297f479e5f7.png">

Then back to the first view (this should have looked like that when loading the page for the first time as the filter is saved in the view):
<img width="934" alt="PS 2021-09-23 à 14 53 36" src="https://user-images.githubusercontent.com/2117580/134510186-f80dd1f2-6956-4880-812f-11f7c277ee3d.png">

**Comment by seanmtaylor** [comment]
Same here, also hiding cards based on a label.

```
-label:frontend -status:Planning
```

It does only seem to be the label which isn't working as the cards with the `Planning` status don't show.

**Comment by azenMatt** [comment]
Hi @armandgrillet @seanmtaylor - it looks like we have something strange going on with labels right now... we are investigating!

**Comment by glortho** [comment]
👋🏼 Hello @armandgrillet @seanmtaylor. If you switch to table view for those boards, is the label column showing or hidden? If hidden, try showing it, returning to board view, and trying again. Let me know if that changes anything, or if the issue remains.

**Reply by seanmtaylor** [reply]
@glortho - Yes that has solved it by adding the Label as a Visible Field on the table view..

**Reply by glortho** [reply]
Thank you for the confirmation @seanmtaylor. The more durable fix for this will ship next week. Sorry for the inconvenience!

**Reply by seanmtaylor** [reply]
No bother @glortho, thanks for the help.

**Comment by azenMatt** [comment, marked as answer]
We _just_ released a fix for this, the problem was that if there is a filter applied - we make sure to load the values even if the field is hidden. But that wasn't working when the filter was a `-`. Thank you @glortho! 

Let me know if this is still giving you any problems

**Reply by armandgrillet** [reply]
It now works for me, thanks for the fix!

**Reply by seanmtaylor** [reply]
Same here, thanks for fixing @azenMatt
